nilearn/plotting/surf_plotting.py

```diff
@@ -3,18 +3,440 @@
+    n_vertices, n_points, img_dim = sample_locations.shape
+    masked = _masked_indices(np.vstack(sample_locations), img_shape, mask=mask)
+    sample_locations = np.rollaxis(sample_locations, -1)
+    sample_indices = np.ravel_multi_index(
+        sample_locations, img_shape, mode='clip').ravel()
+    row_indices, _ = np.mgrid[:n_vertices, :n_points]
+    row_indices = row_indices.ravel()
+    row_indices = row_indices[~masked]
+    sample_indices = sample_indices[~masked]
+    weights = np.ones(len(row_indices))
+    proj = sparse.csr_matrix(
+        (weights, (row_indices, sample_indices.ravel())),
+        shape=(n_vertices, np.prod(img_shape)))
+    proj = sklearn.preprocessing.normalize(proj, axis=1, norm='l1')
+    return proj
+
+
+def _nearest_voxel_sampling(images, mesh, affine, kind='ball', radius=3.,
+                            n_points=None, mask=None):
+    """In each image, measure the intensity at each node of the mesh.
+
+    Image intensity at each sample point is that of the nearest voxel.
+    A 2-d array is returned, where each row corresponds to an image and each
+    column to a mesh vertex.
+    See documentation of niimg_to_surf_data for details.
+
+    """
+    proj = projection_matrix(
+        mesh, affine, images[0].shape, kind=kind, radius=radius,
+        n_points=n_points, mask=mask)
+    data = np.asarray(images).reshape(len(images), -1).T
+    texture = proj.dot(data)
+    texture[np.asarray(proj.sum(axis=1) == 0).ravel()] = np.nan
+    return texture.T
+
+
+def _interpolation_sampling(images, mesh, affine, kind='ball', radius=3,
+                            n_points=None, mask=None):
+    """In each image, measure the intensity at each node of the mesh.
+
+    Image intensity at each sample point is computed with trilinear
+    interpolation.
+    A 2-d array is returned, where each row corresponds to an image and each
+    column to a mesh vertex.
+    See documentation of niimg_to_surf_data for details.
+
+    """
+    sample_locations = _sample_locations(
+        mesh, affine, kind=kind, radius=radius, n_points=n_points)
+    n_vertices, n_points, img_dim = sample_locations.shape
+    grid = [np.arange(size) for size in images[0].shape]
+    interp_locations = np.vstack(sample_locations)
+    masked = _masked_indices(interp_locations, images[0].shape, mask=mask)
+    # loop over images rather than building a big array to use less memory
+    all_samples = []
+    for image in images:
+        interpolator = interpolate.RegularGridInterpolator(
+            grid, image,
+            bounds_error=False, method='linear', fill_value=None)
+        samples = interpolator(interp_locations)
+        samples[masked] = np.nan
+        all_samples.append(samples)
+    all_samples = np.asarray(all_samples)
+    all_samples = all_samples.reshape((len(images), n_vertices, n_points))
+    texture = np.nanmean(all_samples, axis=2)
+    return texture
+
+
+def niimg_to_surf_data(image, surf_mesh,
```

Comment:
In all plotting functions, we said image->img if it not specific to statistical image. Otherwise, stat_map_img.